sdk/applicationinsights/azure-resourcemanager-applicationinsights/src/test/java/com/azure/resourcemanager/applicationinsights/ApplicationInsightsManagerTests.java

```diff
@@ -0,0 +1,131 @@
+
+        applicationInsightsManager = ApplicationInsightsManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        logAnalyticsManager = LogAnalyticsManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testCreateComponent() {
+        ApplicationInsightsComponent component = null;
+        String randomPadding = randomPadding();
+        try {
+            String componentName = "component" + randomPadding;
+            String spaceName = "space" + randomPadding;
+            // @embedStart
+            component = applicationInsightsManager.components()
+                .define(componentName)
+                .withRegion(REGION)
+                .withExistingResourceGroup(resourceGroupName)
+                .withKind("web")
+                .withApplicationType(ApplicationType.WEB)
+                .withWorkspaceResourceId(
+                    logAnalyticsManager.workspaces()
+                        .define(spaceName)
+                        .withRegion(REGION)
+                        .withExistingResourceGroup(resourceGroupName)
+                        .withSku(new WorkspaceSku().withName(WorkspaceSkuNameEnum.PER_GB2018))
+                        .withFeatures(new WorkspaceFeatures().withEnableLogAccessUsingOnlyResourcePermissions(true))
+                        .withWorkspaceCapping(new WorkspaceCapping().withDailyQuotaGb(-1D))
+                        .withRetentionInDays(30)
+                        .withPublicNetworkAccessForIngestion(com.azure.resourcemanager.loganalytics.models.PublicNetworkAccessType.ENABLED)
+                        .withPublicNetworkAccessForQuery(com.azure.resourcemanager.loganalytics.models.PublicNetworkAccessType.ENABLED)
+                        .create()
+                        .id()
```

Comment:
It is better not to create resource nested here.

Just have a line before `applicationInsightsManager` to do this.

Reply:
Fixed in the new version.